registry/endpoint_v1.go

```diff
@@ -42,7 +42,11 @@ func newV1Endpoint(index *registry.IndexInfo, headers http.Header) (*v1Endpoint,
 		return nil, err
 	}
 
-	endpoint, err := newV1EndpointFromStr(GetAuthConfigKey(index), tlsConfig, headers)
+	indexURL := index.Name
```

Comment:
This change deserves an explanation at least in the commit message

Reply:
hm, yes, I should've written that down when I did  the original work. I think this may have been because the further move resulted in a circular import otherwise, but it may have been just to make it clearer what we're doing.

I should actually check how the code is used, ISTR I recently stumbled into the code using it, and we may actually never be sending any authentication information to this endpoint, so perhaps this is even "dead code".

Let me try dig a bit further.

Reply:
OK; I dropped that commit for now, and will look into that with the "move"; there's some other parts that are janky to move. While looking at it, I found some other parts of the code that are only used for search, so moved those as well in the last commit.